lib/src/model/broadcast/broadcast_round_controller.dart

```diff
@@ -72,20 +62,45 @@ class BroadcastRoundController extends _$BroadcastRoundController {
       },
     );
 
-    final round = await ref.withClient(
-      (client) => BroadcastRepository(client).getRound(broadcastRoundId),
+    return BroadcastRoundState(round: round.round, games: round.games, observedGames: ISet());
+  }
+
+  Future<void> _connectSocket({required int version, bool forceReconnect = false}) async {
+    final socketPool = ref.watch(socketPoolProvider);
```

Comment:
Don't use ref.watch outside of widget.build or provider build() methods. Here using ref.read is fine since the socket pool is a singleton.